pipeline_steps/common.groovy

```diff
@@ -372,9 +372,23 @@ def archive_artifacts(Map args = [:]){
       }
 
       pubcloud.uploadArtifacts("artifact_types": artifact_types)
-      if(fileExists(file: "artifact_public_url")){
-        artifact_public_url = readFile(file: "artifact_public_url")
-        currentBuild.description = "<h2><a href='"+artifact_public_url+"'>Build Artifacts</a></h2>"
+      def buildArtifacts
+      if (fileExists("build_artifacts.yml")){
+        buildArtifacts = readYaml file: "build_artifacts.yml"
+      } else{
+        buildArtifacts = ["artifacts": []]
+      }
+      println "Uploaded artefact details:\n${buildArtifacts}"
+      buildArtifacts.artifacts.each {k, v ->
+        // rpc-gating is skipped because it is not a component and so that unit tests can continue
+        if (k == "file" && v.container_name != "rpc-gating"){
+          addArtifactTypeToComponent(v.container_name, v.name, k, v.public_url, "RE")
```

Comment:
Does the public url change with every build? Could we check if the existing value is the same and shortcut this call?

Reply:
The assumption I've made is that only new and unique artefact stores are added given that's all that seems to be required. It can then be extended as the need arises. That said, I did forget to check if it was added so I'll fix that.